sdk/src/utils/hash_utils.rs

```diff
@@ -350,6 +432,24 @@ pub fn verify_asset_by_alg(
         false
     }
 }
+
+pub fn verify_stream_by_alg<R>(
+    alg: &str,
+    hash: &[u8],
+    reader: &mut R,
+    hash_range: Option<Vec<HashRange>>,
+    is_exclusion: bool,
+) -> bool
+where
+    R: Read + Seek + ?Sized,
```

Comment:
Should we be using this instead of CAIRead, at least on any public interface?

Reply:
This used by WASM.  I will change once we have WASM stream support